app/lib/actions/katello/content_view_puppet_module/destroy.rb

```diff
@@ -10,8 +10,9 @@ def plan(repository)
           # puppet modules that are referring to it
           repository.puppet_modules.each do |puppet_module|
             # first, process content view puppet modules that have been specified by version
-            library_repos = ::Katello::Repository.in_environment(repository.organization.library).
-                                                  where(:pulp_id => puppet_module.repoids)
+            library_repos = ::Katello::Repository.
+              in_environment(repository.organization.library).
+              where(:pulp_id => puppet_module.repository_ids)
```

Comment:
puppet_module.repository_ids will return active record ids instead of pulp ids. 

Off the top of my head, I can think of two options to return pulp ids:
1) use something like ```where(:pulp_id => Katello::Respository.where(puppet_module.repository_ids).map(&:pulp_id))```
2) Make a repository_pulp_ids methods for a puppet module